src/lib/broadcast.jl

```diff
@@ -164,31 +164,53 @@ end
 # Avoid hitting special cases for `Adjoint` etc.
 _broadcast(f::F, x...) where F = materialize(broadcasted(f, x...))
 
-_get(x::Tuple, i) = x[i]
-_get(::Nothing, i) = nothing
 collapse_nothings(xs::AbstractArray{Nothing}) = nothing
 collapse_nothings(xs) = xs
 
-@adjoint function broadcasted(::AbstractArrayStyle, f, args...)
+_purefun(::Type{F}) where {F<:Function} = isempty(fieldnames(F))
```

Comment:
```
fieldcount(F) === 0
```
is an alternative used in https://github.com/JuliaDiff/ChainRules.jl/pull/441/files for this, don't think it makes any performance differnce though

Reply:
TIL, that looks a bit tidier. But more importantly, do you think this is safe? I was trying to invent edge cases...

Reply:
edge cases are here, they are pretty weird
https://github.com/JuliaDiff/ChainRules.jl/pull/441/files#r654366514